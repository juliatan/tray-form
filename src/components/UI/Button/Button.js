import React from 'react';
import classes from './Button.module.scss';

const button = (props) => (
  <button {...props} disabled={props.disabled} className={classes.Button}>
    {props.children}
  </button>
);

export default button;
