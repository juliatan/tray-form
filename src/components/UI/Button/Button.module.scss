@import '../../../assets/stylesheets/colours';

.Button {
  display: block;
  width: 96px;
  margin: 16px 0 0 auto;
  padding: 8px 16px;
  color: $white;
  font-size: 1rem;
  line-height: 1.5rem;
  border: 1px solid transparent;
  border-radius: 4px;
  transition: color 0.15s ease-in-out, background-color 0.15s ease-in-out,
    border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
  background-color: $button-success;
  border-color: $button-success;

  &:hover:not(:disabled) {
    background-color: $button-success-hover;
    border-color: $button-success-hover;
  }

  &:disabled {
    opacity: 0.65;
  }

  &:not(:disabled) {
    cursor: pointer;
  }
}
