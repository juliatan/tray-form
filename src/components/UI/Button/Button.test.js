import React from 'react';
import { render } from '@testing-library/react';
import Button from './Button';

describe('Button component', () => {
  it('should render', () => {
    const { asFragment } = render(<Button>Submit</Button>);
    expect(asFragment(<Button>Submit</Button>)).toMatchSnapshot();
  });
});
