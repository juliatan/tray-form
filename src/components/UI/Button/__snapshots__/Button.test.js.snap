// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should render 1`] = `
<DocumentFragment>
  <button
    class="Button"
  >
    Submit
  </button>
</DocumentFragment>
`;
